src/hooks/useCheckVersion/useCheckVersion.ts

```diff
@@ -0,0 +1,60 @@
+import { useContext } from "react";
+import { CampaignConfigContext } from "../../context/campaignConfig";
+import * as config from "../../config";
+
+type CheckVersionResult = "OK" | "OUTDATED_BINARY" | "OUTDATED_BUILD";
+
+export const useCheckVersion = (): (() => CheckVersionResult) => {
+  const { features } = useContext(CampaignConfigContext);
+
+  const isBinaryVersionOK = (): boolean => {
+    const currentBinaryVersion = config.APP_BINARY_VERSION;
+    if (!currentBinaryVersion) {
+      throw new Error("Current app version is not configured properly");
+    }
+
+    const minBinaryVersion = features?.minAppBinaryVersion;
+    if (!minBinaryVersion) {
+      throw new Error("Campaign config not loaded");
+    }
+
+    const currentBinaryVersionSplit = currentBinaryVersion
+      .split(".")
+      .map(Number);
+    const minBinaryVersionSplit = minBinaryVersion.split(".").map(Number);
+    for (let i = 0; i < minBinaryVersionSplit.length; i++) {
+      if (currentBinaryVersionSplit[i] < minBinaryVersionSplit[i]) {
+        return false;
+      } else if (currentBinaryVersionSplit[i] > minBinaryVersionSplit[i]) {
+        return true;
```

Comment:
just clarifying, where will the binary version will be documented? (for us to track haha)

Reply:
Oh we can look at the releases on GitHub, each release creates a new binary version and build version, and also immediately publishes on the corresponding release channel.

However, we don't need to upload a new binary on every single release since the OTA from the publishing will provide the latest build version.